docs/guide/core-concepts.md

```diff
@@ -0,0 +1,427 @@
+|-------------|:----------------:|:------------------------------:|:---------------------------------:|
+| BitCells    | `BitCellType`    | N/A                            | N/A                               |
+| ByteCells   | `ByteCellType`   | `ByteConstantNoDataCellType`   | `ByteUserDefinedNoDataCellType`   |
+| UbyteCells  | `UByteCellType`  | `UByteConstantNoDataCellType`  | `UByteUserDefinedNoDataCellType`  |
+| ShortCells  | `ShortCellType`  | `ShortConstantNoDataCellType`  | `ShortUserDefinedNoDataCellType`  |
+| UShortCells | `UShortCellType` | `UShortConstantNoDataCellType` | `UShortUserDefinedNoDataCellType` |
+| IntCells    | `IntCellType`    | `IntConstantNoDataCellType`    | `IntUserDefinedNoDataCellType`    |
+| FloatCells  | `FloatCellType`  | `FloatConstantNoDataCellType`  | `FloatUserDefinedNoDataCellType`  |
+| DoubleCells | `DoubleCellType` | `DoubleConstantNoDataCellType` | `DoubleUserDefinedNoDataCellType` |
+
+The above table lists `CellType` `DataType`s in the leftmost column
+and `NoData` policies along the top row.
+A couple of points are worth
+making here:
+
+1. Bits are incapable of representing on, off, *and* some `NoData`
+   value. As a consequence, there is no such thing as a Bit-backed tile
+   which recognizes `NoData`.
+2. While the types in the 'No NoData' and 'Constant NoData' are simply
+   singleton objects that are passed around alongside tiles, the greater
+   configurability of 'User Defined NoData' `CellType`s means that they
+   require a constructor specifying the value which will count as
+   `NoData`.
+
+Let's look to how this information can be used:
+```scala
+/** Here's an array we'll use to construct tiles */
+val myData = Array(42, 1, 2, 3)
+
+/** The GeoTrellis-default integer CellType */
+val defaultCT = IntConstantNoDataCellType
+val normalTile = IntArrayTile(myData, 2, 2, defaultCT)
+
+/** A custom, 'user defined' NoData CellType for comparison; we will
+treat 42 as `NoData` for this one */
+val customCellType = IntUserDefinedNoDataValue(42)
+val customTile = IntArrayTile(myData, 2, 2, customCellType)
+
+/** We should expect that the first tile has the value 42 at (0, 0)
+because Int.MinValue is the GeoTrellis-default `NoData` value for
+integers */
+assert(normalTile.get(0, 0) == 42)
+assert(normalTile.getDouble(0, 0) == 42.0)
+
+/** Here, the result is less obvious. Under the hood, GeoTrellis is
+inspecting the value to be returned at (0, 0) to see if it matches our
+custom `NoData` policy and, if it matches (it does), return Int.MinValue
+(no matter your underlying type, `get` on a tile will return an `Int`
+and `getDouble` will return a `Double`.
+
+The use of Int.MinValue and Double.NaN is a result of those being the
+GeoTrellis-blessed values for NoData - below, you'll find a chart that
+lists all such values in the rightmost column */
+assert(customTile.get(0, 0) == Int.MinValue)
+assert(customTile.getDouble(0, 0) == Double.NaN)
+```
```

Comment:
I was shocked to learn that user-defined NODATA values are converted to standard NODATA values by `get`, `set`, and `map` (and what else?) -- e.g. that `customTile.get(0, 0)` returns `Int.MinValue` rather than 42. I think that is likely to surprise most users -- consider covering it specifically in the text rather than incidentally in the example.